crates/optimism/node/tests/e2e/p2p.rs

```diff
@@ -43,13 +43,19 @@ async fn can_sync() -> eyre::Result<()> {
         .update_optimistic_forkchoice(canonical_chain[tip_index - reorg_depth])
         .await?;
     second_node
-        .wait_block((tip - reorg_depth) as u64, canonical_chain[tip_index - reorg_depth], true)
+        .wait_block((tip - reorg_depth) as u64, canonical_chain[tip_index - reorg_depth], false)
         .await?;
     second_node.engine_api.canonical_stream.next().await.unwrap();
 
-    // On third node, sync optimistically up to block number 90a
+    // Trigger backfil sync until block 80
```

Comment:
This seems to be an edge case for new engine. If we sync to the tip via backfill sync and then a reorg happens, we're not going to be able to process it because we will be missing trie updates in in-memory state.

I guess this is unlikely because we'd mostly use backfill sync to reach finalized block